dbms/src/Storages/DeltaMerge/File/DMFile.cpp

```diff
@@ -920,10 +938,16 @@ DMFile::MetaBlockHandle DMFile::writeMergedSubFilePosotionsToBuffer(WriteBuffer
 void DMFile::finalizeMetaV2(WriteBuffer & buffer)
 {
     auto tmp_buffer = WriteBufferFromOwnString{};
-    std::array meta_block_handles = {
+    std::array meta_block_handles = { //
         writeSLPackStatToBuffer(tmp_buffer),
         writeSLPackPropertyToBuffer(tmp_buffer),
+#if 1
         writeColumnStatToBuffer(tmp_buffer),
+#else
+        // ExtendColumnStat is not enabled yet because it cause downgrade compatibility, wait
+        // to be released with other binary format changes.
+        writeExtendColumnStatToBuffer(tmp_buffer),
```

Comment:
How about we write the data at the end (and keep old format)? In this way it may be possible to keep downgrade compatibility as new format will be regarded as extra data and will be discarded.

Reply:
Good suggestion! The ColumnStat is small enough to ignore its impact on performance and data size. We can keep it until we actually need an incompatible storage format changed by other components

Reply:
After some tests, I found the downgrade compatibility can not be satisfied because there is a default branch 😂

https://github.com/pingcap/tiflash/blob/d9dc96bdf7f062ed79459f21143e09e731b4e1ef/dbms/src/Storages/DeltaMerge/File/DMFile.cpp#L1055-L1059

Reply:
Good idea. What kind of downgrade compatibility do we need to have? How about we first land the code that will not throw error (and will not change the format at all) for a few major versions, and then land the compatible format? In this way we will have downgrade compatibility for 1 major version.

Reply:
There is only downgrade compatibility requirement between patch versions under the same minor version. And no requirement across minor or major versions. But it is good to have.

I've removed the exception thrown branch in this PR because the correctness of bytes read is protected by checksum. I think ignoring unknown meta block types without exception or logging is an acceptable behavior.

@JinheLin @breezewish